test/suites/integration/msp/debt-collection.test.ts

```diff
@@ -0,0 +1,254 @@
+      }
+
+      strictEqual(bucketRootChangedDataBlob.newRoot.toString(), local_bucket_root.toString());
+
+      const isFileInForest = await mspApi.rpc.storagehubclient.isFileInForest(
+        newBucketEventDataBlob.bucketId.toString(),
+        acceptedFileKey
+      );
+
+      invariant(isFileInForest.isTrue, "File is not in forest");
+
+      // Seal block containing the MSP's transaction response to the storage request
+      await userApi.wait.mspResponseInTxPool();
+      await userApi.sealBlock();
+
+      const fileKeys2: string[] = [];
+
+      const mspAcceptedStorageRequestEvents = await userApi.assert.eventMany(
+        "fileSystem",
+        "MspAcceptedStorageRequest"
+      );
+
+      for (const e of mspAcceptedStorageRequestEvents) {
+        const mspAcceptedStorageRequestDataBlob =
+          userApi.events.fileSystem.MspAcceptedStorageRequest.is(e.event) && e.event.data;
+        if (mspAcceptedStorageRequestDataBlob) {
+          fileKeys2.push(mspAcceptedStorageRequestDataBlob.fileKey.toString());
+        }
+      }
+
+      invariant(fileKeys2.length === 2, "Expected 2 file keys");
+
+      // Allow time for the MSP to update the local forest root
+      await sleep(3000);
+
+      const local_bucket_root2 = await mspApi.rpc.storagehubclient.getForestRoot(
+        newBucketEventDataBlob.bucketId.toString()
+      );
+
+      const { event: bucketRootChangedEvent2 } = await userApi.assert.eventPresent(
+        "providers",
+        "BucketRootChanged"
+      );
+
+      const bucketRootChangedDataBlob2 =
+        userApi.events.providers.BucketRootChanged.is(bucketRootChangedEvent2) &&
+        bucketRootChangedEvent2.data;
+
+      if (!bucketRootChangedDataBlob2) {
+        throw new Error("Expected BucketRootChanged event but received event of different type");
+      }
+
+      strictEqual(bucketRootChangedDataBlob2.newRoot.toString(), local_bucket_root2.toString());
+
+      for (const fileKey of fileKeys2) {
+        const isFileInForest = await mspApi.rpc.storagehubclient.isFileInForest(
+          newBucketEventDataBlob.bucketId.toString(),
+          fileKey
+        );
+        invariant(isFileInForest.isTrue, "File is not in forest");
+      }
+    });
+
+    it("MSP is charging user", async () => {
+      // Calculate how many blocks to advance until next challenge tick.
+      const currentBlock = await userApi.rpc.chain.getBlock();
+      const currentBlockNumber = currentBlock.block.header.number.toNumber();
+
+      const blocksToAdvance = currentBlockNumber + (12 - (currentBlockNumber % 12)) + 1; // 12 is the msp_charging_freq that we setup
+      if (blocksToAdvance > currentBlockNumber) {
+        await userApi.advanceToBlock(blocksToAdvance);
+      }
+
+      // Verify that the MSP charged the users after the notified
+      await userApi.assert.eventPresent("paymentStreams", "PaymentStreamCharged");
```

Comment:
We should probably verify the amount charged to make sure it is correct.

Reply:
Agreed